tsl/test/expected/hypercore_dump_restore.out

```diff
@@ -0,0 +1,310 @@
+select create_hypertable('hyperdump', by_range('time'), create_default_indexes => false);
+NOTICE:  adding not-null constraint to column "time"
+ create_hypertable 
+-------------------
+ (1,t)
+(1 row)
+
+insert into hyperdump (time, device, tempc)
+select t, ceil(random()*10), random()*60
+from generate_series('2022-06-01'::timestamptz, '2022-07-01', '5m') t;
+create index time_device_idx on hyperdump (device, time desc);
+alter table hyperdump set (
+      timescaledb.compress_orderby='time',
+      timescaledb.compress_segmentby='device');
+\set TEST_BASE_NAME hypercore_pgdump
+SELECT
+    format('%s/results/%s_results_original.out', :'TEST_OUTPUT_DIR', :'TEST_BASE_NAME') AS "TEST_RESULTS_ORIGINAL",
+    format('%s/results/%s_results_restored.out', :'TEST_OUTPUT_DIR', :'TEST_BASE_NAME') AS "TEST_RESULTS_RESTORED" \gset
+-- Save uncompressed table query output as a reference to compare against
+\o :TEST_RESULTS_ORIGINAL
+select * from hyperdump order by time, device;
+\o
+explain (costs off)
+select * from hyperdump where device = 2 and time < '2022-06-03';
+                                                 QUERY PLAN                                                 
+------------------------------------------------------------------------------------------------------------
+ Append
+   ->  Index Scan using _hyper_1_1_chunk_time_device_idx on _hyper_1_1_chunk
+         Index Cond: ((device = 2) AND ("time" < 'Fri Jun 03 00:00:00 2022 PDT'::timestamp with time zone))
+   ->  Index Scan using _hyper_1_2_chunk_time_device_idx on _hyper_1_2_chunk
+         Index Cond: ((device = 2) AND ("time" < 'Fri Jun 03 00:00:00 2022 PDT'::timestamp with time zone))
+(5 rows)
+
+-- Convert to hypercore
+select compress_chunk(ch, hypercore_use_access_method=>true) from show_chunks('hyperdump') ch;
+             compress_chunk             
+----------------------------------------
+ _timescaledb_internal._hyper_1_1_chunk
+ _timescaledb_internal._hyper_1_2_chunk
+ _timescaledb_internal._hyper_1_3_chunk
+ _timescaledb_internal._hyper_1_4_chunk
+ _timescaledb_internal._hyper_1_5_chunk
+ _timescaledb_internal._hyper_1_6_chunk
+(6 rows)
+
+reindex table hyperdump;
+explain (costs off)
+select * from hyperdump where device = 2 and time < '2022-06-03';
+                                                 QUERY PLAN                                                 
+------------------------------------------------------------------------------------------------------------
+ Append
+   ->  Index Scan using _hyper_1_1_chunk_time_device_idx on _hyper_1_1_chunk
+         Index Cond: ((device = 2) AND ("time" < 'Fri Jun 03 00:00:00 2022 PDT'::timestamp with time zone))
+   ->  Index Scan using _hyper_1_2_chunk_time_device_idx on _hyper_1_2_chunk
+         Index Cond: ((device = 2) AND ("time" < 'Fri Jun 03 00:00:00 2022 PDT'::timestamp with time zone))
+(5 rows)
+
+set timescaledb.hypercore_copy_to_behavior='all_data';
```

Comment:
Do you have tests to set this to a nonsensical value?

Reply:
Added